### PR TITLE
UGENE-7891 Add additional filter parameters

### PR DESCRIPTION
**Pull request by DmitriiSukhomlinov**
I will change source branch after [UGENE-7891](https://github.com/ugeneunipro/ugene/pull/1314) will be merged

### CODE REVIEW

File: src/plugins/enzymes/src/EnzymesIO.cpp
Comment (yalgaer) [inline src/plugins/enzymes/src/EnzymesIO.cpp]: nit: 
`d->nondegenerate = d->alphabet->getId() == BaseDNAAlphabetIds::NUCL_DNA_DEFAULT() && !d->seq.contains(EnzymeData::UNDEFINED_BASE);` 
-> 
`d->nondegenerate = d->alphabet->getId() == BaseDNAAlphabetIds::NUCL_DNA_DEFAULT() && d->uninterrupted;` ?

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.cpp:623]: `cbShowNondegenerate` automatically includes `cbShowUninterrupted` results.

What do you think if we make `cbShowNondegenerate` enabled & checked only if 'cbShowUninterrupted' is checked?

**Review by yalgaer — COMMENTED** [review]
Looks good.
Will approve after final merges & UI layout check.

File: src/plugins/enzymes/src/EnzymesIO.cpp
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/EnzymesIO.cpp]: OK

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesDialog.cpp:623]: No, my bad, I checked SnapGene again and found out, that "Nondegenerate" enzymes do not contain external DNA alphabet, but could contain Ns. I will rework it.

File: src/plugins/enzymes/src/EnzymesIO.cpp
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/EnzymesIO.cpp]: See comment below

File: src/plugins/enzymes/src/FindEnzymesDialog.ui
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.ui]: Typo in the text

**Review by yalgaer — CHANGES_REQUESTED** [review]
The current UI looks like something is not shown by default and will be added only if the checkbox is checked:
image

I suggest to rework it so a more compact (no repetitive text, move it to the tooltip if needed) and more clear label: 
image

**Comment by DmitriiSukhomlinov** [comment]
Fixed

**Review by yalgaer — APPROVED** [review]

**Pull request merged**
